plugins/amd-kria/fu-amd-kria-image-reg.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2024 Advanced Micro Devices Inc.
+ * All rights reserved.
+ *
+ * This file is provided under a dual MIT/LGPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ * AMD Chooses the MIT license part of Dual MIT/LGPLv2 license agreement.
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR MIT
+ */
+
+#include "config.h"
+
+#include "fu-amd-kria-image-reg.h"
+
+struct _FuAmdKriaImageFirmware {
+	FuFirmwareClass parent_instance;
+};
+
+#define VERSION_OFFSET	0x70
+#define VERSION_SIZE	0x24
+
+G_DEFINE_TYPE(FuAmdKriaImageFirmware, fu_amd_kria_image_firmware, FU_TYPE_FIRMWARE)
+
+static gboolean
+fu_amd_kria_image_firmware_parse(FuFirmware *firmware,
+			         GInputStream *stream,
+			         gsize offset,
+			         FwupdInstallFlags flags,
+			         GError **error)
+{
+	const guint8 *buf;
+	gsize bufsz = 0;
+	g_autoptr(GBytes) fw = NULL;
+	g_autofree gchar *version = NULL;
+
+	fw = fu_input_stream_read_bytes(stream, offset, G_MAXSIZE, error);
+	if (fw == NULL)
+		return FALSE;
```

Comment:
seems wasteful to read the entire firmware to just set the version. Can we read from `offset+VERSION_OFFSET` for length `VERSION_SIZE` instead? Then we can just `fu_strsafe` the buffer.